### PR TITLE
Migrations

### LINKED ISSUE #21: addColumnToTable: Users
- [x] add style-preferences column
- [x] possibly an array of style-type strings

### LINKED ISSUE #22: addColumnsToTable: Subscriptions
- [x] add style-type column
- [x] this will be an array of image ID's?

### LINKED ISSUE #23: addColumnsToTable: Products
- [x] add style-type column
- [x] this will an array of image ID's?

### PR DESCRIPTION
**Pull request by stillsheryl**
#### What does  this PR do?
- Adds migrations for Users, Products & Subscriptions. This added the following columns:
  * Users has a stylePreferences column; the data is stored as TEXT
  * Products has a styleType column; the data is stored as a STRING
  * Subscriptions has a styleType column; the data is stored as a STRING
#### Where should the reviewer start?
- Review the added columns to each table by viewing the files changed.
#### How should this be manually tested?
- Run the servers for both web and API and you will be able to navigate the pages.
#### Questions:
  - Do Migrations Need to be ran? 
      - Yes! run the `sequelize db:migrate` command.
  - Do Environment Variables need to be set? 
      - No
  - Any other deploy steps?
      - No

- closes #21 
- closes #22 
- closes #23

### CODE REVIEW

File: code/api/package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "@babel/node": "7.10.5",
     "@babel/plugin-transform-runtime": "7.11.5",
     "@babel/preset-env": "7.11.5",
-    "nodemon": "2.0.4"
+    "nodemon": "2.0.4",
+    "sequelize-cli": "^5.5.1"
```
Comment (dcoleman21) [inline code/api/package.json:27]: This was added with the npm install --save-dev sequelize-cli command

File: code/api/src/modules/user/model.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     role: {
       type: DataTypes.TEXT
+    },
+    stylePreference: {
+      type: DataTypes.TEXT
```
Comment (stillsheryl) [inline code/api/src/modules/user/model.js:19]: We chose TEXT here to hold the string or potential array of strings if a user has multiple subscriptions.

File: code/api/package.json [hunk 3]
```diff
@@ -40,7 +41,6 @@
     "multer": "1.4.2",
     "mysql2": "2.1.0",
     "pg": "^8.5.1",
-    "sequelize": "5.21.12",
-    "sequelize-cli": "5.5.1"
+    "sequelize": "^5.21.12"
```
Comment (dcoleman21) [inline code/api/package.json:44]: Same as above comment.

File: code/api/package.json [hunk 4]
```diff
@@ -23,7 +23,8 @@
     "@babel/node": "7.10.5",
     "@babel/plugin-transform-runtime": "7.11.5",
     "@babel/preset-env": "7.11.5",
-    "nodemon": "2.0.4"
+    "nodemon": "2.0.4",
+    "sequelize-cli": "^5.5.1"
   },
```
Comment (s-steel) [inline code/api/package.json:28]: Installed sequelize local using commands in the slack channel.

File: code/api/src/migrations/20210208174208-add_stylePreference_to_users.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208174208-add_stylePreference_to_users.js:4]: This section was manually added once the migration was created.

File: code/api/src/migrations/20210208174208-add_stylePreference_to_users.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    return queryInterface.addColumn(
+      'users',
+      'stylePreference',
+     Sequelize.TEXT
+    );
+
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    return queryInterface.removeColumn(
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208174208-add_stylePreference_to_users.js:14]: This section was manually added once the migration was created.

File: code/api/src/migrations/20210208181640-add_styleType_to_subscriptions.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208181640-add_styleType_to_subscriptions.js:4]: This section was manually added once the migration was created.

File: code/api/src/migrations/20210208181640-add_styleType_to_subscriptions.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    return queryInterface.addColumn(
+      'subscriptions',
+      'styleType',
+     Sequelize.TEXT
+    );
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    return queryInterface.removeColumn(
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208181640-add_styleType_to_subscriptions.js:13]: This section was manually added once the migration was created.

File: code/api/src/migrations/20210208182658-add_styleType_to_products.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208182658-add_styleType_to_products.js:4]: This section was manually added once the migration was created.

File: code/api/src/migrations/20210208182658-add_styleType_to_products.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    return queryInterface.addColumn(
+      'products',
+      'styleType',
+     Sequelize.TEXT
+    );
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    return queryInterface.removeColumn(
```
Comment (dcoleman21) [inline code/api/src/migrations/20210208182658-add_styleType_to_products.js:13]: This section was manually added once the migration was created.

File: code/api/src/modules/product/model.js [hunk 11]
```diff
@@ -20,6 +20,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     image: {
       type: DataTypes.TEXT
+    },
+    styleType: {
```
Comment (dcoleman21) [inline code/api/src/modules/product/model.js:24]: This was manually added once the migration was made.

File: code/api/src/migrations/20210208174208-add_stylePreference_to_users.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    return queryInterface.addColumn(
+      'users',
+      'stylePreference',
```
Comment (s-steel) [inline code/api/src/migrations/20210208174208-add_stylePreference_to_users.js:7]: stylePreference used for users, while styleType is used for subscriptions and products.

File: code/api/src/modules/product/types.js [hunk 13]
```diff
@@ -14,6 +14,7 @@ const ProductType = new GraphQLObjectType({
     gender: { type: GraphQLInt },
     description: { type: GraphQLString },
     image: { type: GraphQLString },
+    styleType: { type: GraphQLString },
```
Comment (dcoleman21) [inline code/api/src/modules/product/types.js:17]: This was manually added once the migration was made.

File: code/api/src/modules/subscription/model.js [hunk 14]
```diff
@@ -8,6 +8,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     crateId: {
       type: DataTypes.INTEGER
+    },
+    styleType: {
```
Comment (dcoleman21) [inline code/api/src/modules/subscription/model.js:12]: This was manually added once the migration was made.

File: code/api/src/modules/subscription/types.js [hunk 15]
```diff
@@ -14,9 +14,10 @@ const SubscriptionType = new GraphQLObjectType({
     id: { type: GraphQLInt },
     user: { type: UserType },
     crate: { type: CrateType },
+    styleType: { type: GraphQLString },
```
Comment (dcoleman21) [inline code/api/src/modules/subscription/types.js:17]: This was manually added once the migration was made.

File: code/api/src/modules/user/model.js [hunk 16]
```diff
@@ -14,6 +14,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     role: {
       type: DataTypes.TEXT
+    },
+    stylePreference: {
```
Comment (dcoleman21) [inline code/api/src/modules/user/model.js:18]: This was manually added once the migration was made.

File: code/api/src/modules/user/types.js [hunk 17]
```diff
@@ -12,6 +12,7 @@ const UserType = new GraphQLObjectType({
     email: { type: GraphQLString },
     password: { type: GraphQLString },
     role: { type: GraphQLString },
+    stylePreference: { type: GraphQLString },
```
Comment (dcoleman21) [inline code/api/src/modules/user/types.js:15]: This was manually added once the migration was made.

**Review by dcoleman21 — COMMENTED** [review]
I left some comments to help annotate what was done.

File: code/api/src/migrations/20210208182658-add_styleType_to_products.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    return queryInterface.addColumn(
+      'products',
+      'styleType',
```
Comment (s-steel) [inline code/api/src/migrations/20210208182658-add_styleType_to_products.js:7]: styleType not to be confused with 'type' which refers to the clothing type ie cloth, accessories.

**Review by s-steel — APPROVED** [review]

**Review by dcoleman21 — APPROVED** [review]

**Pull request merged**